_posts/2020-07-08-pull-requests-require-context.md

```diff
@@ -0,0 +1,148 @@
+
+### Give a guided tour of the changes
+
+[Devon](https://github.com/dblandin) takes the idea of adding context to individual lines to the next level
+[in this PR](https://github.com/artsy/reaction/pull/2774#pullrequestreview-288095754). He takes advantage of
+markdown to give us a virtual tour of the changes, at each stop providing helpful information and a link to the
+next change. It's like he's sitting next to you!
+
+## Show Your Work
+
+If your PR contains work that is beyond trivial, "showing your work" can help reviewers understand what they're
+reviewing.
+
+### Make small, self-contained commits
+
+A good PR starts with good commits. Good commits are small, self-contained, and leave the codebase always in a
+working state. With good commits, reviewers can see exactly how you worked through the problem you were solving.
+[Here's a PR](https://github.com/artsy/convection/pull/645) from [Jon](https://github.com/jonallured) that
+demonstrates the use of small, self-contained commits to describe his approach to refactoring code before fixing a
+bug.
+
+_Bonus tip_: it can be easier to review PRs with many small commits via the
+[_Commits_](https://github.com/artsy/convection/pull/645/commits) tab instead of the
+[_Files changed_](https://github.com/artsy/convection/pull/645/files) tab.
+
+### Demonstrate the results
+
+Pictures are a worth a thousand words. Animated gifs are worth a thousand pictures (uhhhh, in file size too 😬). An
+animated gif showing the outcome of your PR gives reviewers a demo, and confidence that you've verified your
+changes.
+
+[Here's a PR](https://github.com/artsy/eigen/pull/3039) from [Christina](https://github.com/xtina-starr) that
+includes an animated gif of the changes she's made. Combined with her descriptions of the original issue and the
+fix, I've got a clear idea of what was happening before these changes and what will be happening after.
+
+### Document the unseen
+
+Sometimes a PR's changes have effects outside of the UI. There are still ways to give reviewers proof that the
+changes have the desired effects.
+
+[Here's a PR](https://github.com/artsy/eigen/pull/3206) from [Yuki](https://github.com/yuki24) that not only
+demonstrates what's happening in the UI, but also assures me that the back-end data is getting updated properly
+through a Rails console.
+
+[Christina](https://github.com/xtina-starr) authored [this PR](https://github.com/artsy/reaction/pull/3441) which
+shows the UI changes in addition to some output from her browser console, demonstrating that analytics tracking
+calls are firing correctly.
+
+### Share your progress
+
+One mistake many engineers make with non-trivial pull requests is to wait to open them until they're "done". If
+there are changes you'd like to get people's eyes on quickly, open a PR before the work is done and mark it as
+in-progress with`WIP` in the title. This helps avoid rework by starting early discussions about your approach.
```

Comment:
Or use a "draft" PR.

(Also: minor typo in this line.)

Reply:
There's also the benefit of getting early feedback to validate your approach 👍